index.js

```diff
@@ -103,40 +110,58 @@ module.exports = function(options){
                 return outputBasename + options.suffix(index) + outputExtension;
             };
 
-            var addImports = function(index, contents){
-                // only the first file should have @imports
-                if(!options.imports || index){
-                  return contents;
-                }
-
-                var imports = '';
+            var addImports = function(aFile, fileNamesOfPartsToImport){
                 var parameters = options.cacheBuster ? '?z=' + Math.round((Math.random() * 999)) : '';
-                for (var i = 1; i < numberOfSplits; i++) {
-                    imports += "@import url('" + createBlessedFileName(i) + parameters + "');\n\n";
+                var filePath = path.relative(aFile.base, aFile.path);
+                var concat = new Concat(shouldCreateSourcemaps, filePath, '\n');
+                for (var i = 0; i < fileNamesOfPartsToImport.length; i++) {
+                    concat.add(null, "@import url('" + fileNamesOfPartsToImport[i] + parameters + "');\n");
                 }
 
-                return imports + contents;
-            };
+                concat.add(filePath, aFile.contents, JSON.stringify(aFile.sourceMap));
 
+                aFile.contents = concat.content;
+                if (shouldCreateSourcemaps) {
+                    aFile.sourceMap = JSON.parse(concat.sourceMap);
+                }
+            };
 
             var outputFiles = [];
-            for(var j = numberOfSplits - 1; j >= 0; j--) {
-                var newIndex = numberOfSplits - 1 - j;
-                var outputPath = newIndex
-                    ? path.resolve(path.join(outputPathStart, createBlessedFileName(newIndex)))
-                    : outputFilePath;
+            var nonMasterPartFileNames = [];
+            for(var j = 0; j < numberOfSplits; j++) {
+                var oneBasedIndex = j + 1;
+                var isAtLastElement = j == numberOfSplits - 1;
```

Comment:
This is nitpicking but `var isAtLastElement  = oneBasedIndex === numberOfSplits;` is clearer